src/dlgConnectionProfiles.cpp

```diff
@@ -1016,63 +1047,108 @@ void dlgConnectionProfiles::fillout_form()
+
+    // Now read the stored order - and the size we stored for this dialog:
+    QSize desiredDialogSize{};
+    QStringList existingProfilesOrderedList = loadProfilesOrderAndDlgSize(&desiredDialogSize);
+    // First make a pass through the QMap of QListWidgetItems - and identify
+    // those whose name is NOT in the stored, ordered, list of profiles, so we
+    // can place them first:
+    QMutableMapIterator<QString, QListWidgetItem*> itProfile(nameToItemMap);
+    while (itProfile.hasNext()) {
+        itProfile.next();
+        if (!existingProfilesOrderedList.contains(itProfile.key(), Qt::CaseInsensitive)) {
+            // This is a new one that we don't have an order position for, so
+            // add it straight into the listWiget so it will be above the
+            // "known" ones - and give it a green background to highlight that
+            // it is a new one:
+            auto pItem = itProfile.value();
+            // Remove it from the QMap so there are less to deal with - so the
+            // second pass is simpler:
+            itProfile.remove();
+            // Insert it into the QListWidget
             profiles_tree_widget->addItem(pItem);
-            description = getDescription(qsl("mudlet.org"));
-            if (!description.isEmpty()) {
-                pItem->setToolTip(utils::richText(description));
-            }
+            // Now that it is owned by the QListWidget the setHidden(bool)
+            // method will operate to hide it if requested:
+            pItem->setHidden(pItem->data(csmHiddenRole).toBool());
+            pItem->setBackground(QColorConstants::Green);
         }
-#endif
-    } else {
-        pItem = new QListWidgetItem();
-        for (const QString& onlyShownPredefinedProfile : onlyShownPredefinedProfiles) {
-            auto details = TGameDetails::findGame(onlyShownPredefinedProfile);
-            setupMudProfile(pItem, onlyShownPredefinedProfile, (*details).description, (*details).icon);
+    }
+
+    // Now iterate through the ordered list of profile names to pick out the
+    // items we do have a QListWidgetIrem for, in the stored order:
+    QStringListIterator itOrderedProfile(existingProfilesOrderedList);
+    while (itOrderedProfile.hasNext()) {
+        auto existingProfileName = itOrderedProfile.next();
+        if (nameToItemMap.contains(existingProfileName)) {
+            // Transfer the item from the QMap to the QListWidget
+            auto pItem = nameToItemMap.take(existingProfileName);
+            profiles_tree_widget->addItem(pItem);
+            pItem->setHidden(pItem->data(csmHiddenRole).toBool());
+            // A default constructed brush will clear any existing set colour:
+            pItem->setBackground(QBrush());
         }
     }
 
-    setProfileIcon();
+    // Check whether we've processed ALL the profiles we had created
+    // QListWidgetItems for:
+    if (!nameToItemMap.isEmpty()) {
+        itProfile.toFront();
```

Comment:
I don't think this can happen but it must be a theoretical possibility - so this will handle it.